### PR TITLE
resource/allowed-conversions/<resource_id> endpoint

### PR DESCRIPTION
**Pull request by magdasta**

### CODE REVIEW

File: TranscodingDemo/Server/views/allowed_conversions_view.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+from Server import permissions
+from Server.authorization import AuthenticationException
+from Server.models.resource import Resource
+from Server.transcoding.video_info import VideoInfo
+from django.http import JsonResponse
+from rest_framework.exceptions import PermissionDenied
+from rest_framework.views import APIView
+
+
+class ResourceNotFound(Exception):
```
Comment (nieznanysprawiciel) [inline TranscodingDemo/Server/views/allowed_conversions_view.py]: You could use ResourceUnavailable from exceptions.py

File: TranscodingDemo/Server/views/allowed_conversions_view.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    pass
+
+
+class AllowedConversionsView(APIView):
+    permission_classes = [permissions.IsResourceOwner]
+
+    def get(self, request, resource_id):
+        try:
+            resource = Resource.get(resource_id)
```
Comment (nieznanysprawiciel) [inline TranscodingDemo/Server/views/allowed_conversions_view.py]: Resource has function from_id which will throw ResourceUnavaible.

File: TranscodingDemo/Server/views/allowed_conversions_view.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+        except (PermissionDenied, ResourceNotFound):
+            return JsonResponse(status=HTTPStatus.NOT_FOUND, data={})
+
+        # this isn't expected here
+        except AuthenticationException:
+            return JsonResponse(status=HTTPStatus.FORBIDDEN, data={})
+
+        except Exception as e:
+            return JsonResponse(status=HTTPStatus.INTERNAL_SERVER_ERROR, data={"message": repr(e)})
```
Comment (nieznanysprawiciel) [inline TranscodingDemo/Server/views/allowed_conversions_view.py]: If we catch generic exception, probably we don't want to show user repr(e). But we would like to log repr(e) for our own knowledge.

File: TranscodingDemo/Server/views/allowed_conversions_view.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+            response_data = {
+                "allowedFormats": VideoInfo.get_allowed_formats(),
+                "allowedResolutions": video_info.get_allowed_resolutions(),
+                "conversionsMap": video_info.get_allowed_conversions(),
+                "maxSubtasks": video_info.estimate_max_num_subtasks(),
+                "defaultSubtasksNumber": video_info.get_default_subtasks_number()
+            }
+            return JsonResponse(status=HTTPStatus.OK, data=response_data)
+
+        except (PermissionDenied, ResourceNotFound):
```
Comment (nieznanysprawiciel) [inline TranscodingDemo/Server/views/allowed_conversions_view.py]: We have convention that PermissionDenied returns BAD_REQUEST.

File: TranscodingDemo/Server/views/allowed_conversions_view.py [hunk 5]
```diff
@@ -1,26 +1,26 @@
...
 
 
 class AllowedConversionsView(APIView):
     permission_classes = [permissions.IsResourceOwner]
 
     def get(self, request, resource_id):
         try:
-            resource = Resource.get(resource_id)
+            resource = Resource.from_id(resource_id)
             if resource is None:
```
Comment (nieznanysprawiciel) [inline TranscodingDemo/Server/views/allowed_conversions_view.py]: You don't need this check anymore, because you catch exception

Comment (magdasta) [reply TranscodingDemo/Server/views/allowed_conversions_view.py]: Oh, OK. Get it.

**Pull request merged**